open3d_slam/src/SubmapCollection.cpp

```diff
@@ -253,14 +253,17 @@ const Constraints& SubmapCollection::getOdometryConstraints() const {
 }
 
 Constraints SubmapCollection::buildLoopClosureConstraints(
-		const TimestampedSubmapIds &loopClosureCandidatesIdxs) const {
+		const TimestampedSubmapIds &loopClosureCandidatesIdxs)  {
 	Constraints retVal;
 	for (const auto &id : loopClosureCandidatesIdxs) {
 		const auto constraints = placeRecognition_.buildLoopClosureConstraints(mapToRangeSensor_, *this,
 				adjacencyMatrix_, id.submapId_, activeSubmapIdx_, id.time_);
 		if (!constraints.empty()) {
 			for (int i = 0; i < constraints.size(); ++i) {
 				retVal.push_back(constraints.at(i));
+				const int v1 = constraints.at(i).sourceSubmapIdx_;
+				const int v2 = constraints.at(i).sourceSubmapIdx_;
```

Comment:
Does this constraint add a `self` edge in the loop closure graph? , from constraint `i` to again `i` 

Sorry for the naive question, I am rusty on loop closure

Reply:
Good catch! This is definitely a bug. The way it is implemented now, it does not affect anything, however that was not the intention. I will fix this in a separate PR.